src/clients/firewall_stage1_finish.rb

```diff
@@ -80,6 +91,64 @@ def main
       Builtins.y2milestone("firewall_stage1_finish finished")
       deep_copy(@ret)
     end
+
+    # Returns true if all services are known to firewall
+    # @param [Array <String>] services
+    # @return [Boolean] if all are known
+    def known_firewall_services?(services)
+      @all_known_services ||= SuSEFirewallServices.all_services.keys
+
+      (services - @all_known_services).empty?
+    end
+
+    # Reads and adjust the configuration for SuSEfirewall2 according to the current proposal.
+    def adjust_configuration
+      enable_fw = SuSEFirewall4Network.Enabled1stStage
+      enable_sshd = SuSEFirewall4Network.EnabledSshd
+      open_ssh_port = SuSEFirewall4Network.EnabledSsh1stStage
+      open_vnc_port = SuSEFirewall4Network.EnabledVnc1stStage
+
+      log.info "After installation, firewall will be #{enable_fw ? "enabled" : "disabled"}, " \
+        "SSHD will be #{enable_sshd ? "enabled" : "disabled"}, " \
+        "SSH port will be #{open_ssh_port ? "open" : "closed"}, " \
+        "VNC port will be #{open_vnc_port ? "open" : "closed"}"
+
+      # Read the configuration from sysconfig
+      # bnc#887406: The file is in inst-sys
+      previous_state = Progress.set(false)
+      SuSEFirewall.Read
+      Progress.set(previous_state)
+
+      SuSEFirewall.SetEnableService(enable_fw)
+      SuSEFirewall.SetStartService(enable_fw)
```

Comment:
Do we need to execute the rest of the method if `enable_fw == false` ?

Reply:
No, we don't need the rest and even we should not take care of the service enablement, in that case we should read the profile definition to know what the user wanted not what is the default.

Reply:
But of course it has to fit well with the fact that if the firewall is enabled by default we have to allow the remote installation during the second stage.